src/Auth/DatabaseAuthenticator.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Auth;
+
+use Tempest\Auth\Exceptions\InvalidLoginException;
+use Tempest\Database\Query;
+
+final class DatabaseAuthenticator extends GenericAuthenticator
```

Comment:
I don't like having to extend from other classes. It's better to extract common functionality to a trait and have this class implement the interface directly.

I think, for now, we only need a database authenticator

Reply:
Oh ok, so you’re aiming for Composition instead. Got it, I’ll refactor it